src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -48,6 +51,12 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig, protected va
 
   def createCluster(googleProject: GoogleProject, clusterName: ClusterName, clusterRequest: ClusterRequest): Future[Cluster] = {
     def create() = {
+      // place in processClusters w/ adding label op - not bother now, doing it after Rob's PR merges
+      if ((!clusterRequest.numberOfWorkers.isDefined || clusterRequest.numberOfWorkers.get == 0)
+          && (clusterRequest.workerMachineType.isDefined || clusterRequest.workerDiskSize.isDefined
+              || clusterRequest.numberOfWorkerLocalSsds.isDefined || clusterRequest.numberOfPreemptibleWorkers.isDefined))
+        throw WorkerConfigsSpecifiedForZeroWorkers()
```

Comment:
If the numberOfWorkers is None or 0 and any of the worker properties _are_ defined,  throw an exception. The other option is, in the case of a Single Node cluster, ignore any worker properties in the request

Reply:
I'd say just ignore anything worker-config-y if the number of workers is 0.